airoha: backport additional patch + AS21xxx PHY support

Ansuel opened the pull request:
Backport additional patch merged upstream and also add support for AS21xxx PHY.

Ansuel committed:
airoha: replace thermal patch with upstream version

Replace thermal patch with upstream version. The thermal maintainer
reported that the sysfs entry are considered deprecated and that slope
and offset should be handled internally to the driver.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
firmware: Add support for Aeonsemi AS21xxx firmware

Add support for Aeonsemi AS21xxx firmware blob. Firmware has been
submitted and accepted to linux-firmware. Current version is 1.8.2.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: backport support for Aeonsemi AS21xxx PHY

Backport support for Aeonsemi AS121xxx PHY. The PHY require dedicated
firmware to be loaded to correctly work and support a big family of
Aeonsemi PHY that provide from 1G to 10G speed.

Automatically refresh all affected patch and file (rtl PHY).

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Rondom commented on package/firmware/as21x1x-firmware/Makefile:
```diff
@@ -0,0 +1,32 @@
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=as21x1x-firmware
```
> Add support for Aeonsemi AS21xxx firmware blob. Firmware has been
submitted and accepted to linux-firmware. Current version is 1.8.2.

Why not take it from linux-firmware directly. OpenWrt linux-firmware package is at 20250808 which should include this firmware

Ansuel committed:
firmware: Add support for Airoha EN7581 NPU firmware

Add support for Airoha EN7581 NPU firmware present in linux-firmware.
Support for it is fully upstream with the Ethernet part fully pushed and
the Wireless Offload currently in progress for various WiFi chip.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: reintroduce Mediatek PHY patch to backport directory

Mediatek PHY patch has been merged upstream. Reintroduce them to
backport directory as the same PHY is also needed for Airoha target.

All the affected patch automatically refreshed.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
airoha: backport patch adding support for AN7581 Ethernet PHY

Backport patch adding support for AN7581 Ethernet PHY based on the same
Mediatek embedded Switch PHY.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
airoha: backport trivial fixes for pinctrl and ethernet driver

Backport trivial fixes from upstream related to pinctrl and ethernet
driver.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
airoha: major backport of Airoha Ethernet driver feature support

Major backport of upstream patch for support of multiple feature of the
Airoha Ethernet driver.

Feature backported are TSO, Jumbo packet, Offload and initial Wlan
Offload support.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
airoha: add NPU and reserved memory node for AN7581

Add the NPU and reserved memory node for AN7581 dtsi since it's not
supported.

Link: https://github.com/openwrt/openwrt/pull/19816
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@dangowrt I'm moving the mediatek PHY patches to backport for 6.6 (for 6.12 it was already done somehow). Is it ok for you? It does improve consistency and makes it easier to backport stuff to 24.10.
